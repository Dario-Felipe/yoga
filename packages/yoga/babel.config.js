module.exports = {
  extends: '../../babel.config.js',
  plugins: ['transform-import-css'],
};
