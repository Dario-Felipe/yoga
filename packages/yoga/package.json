{
  "name": "@gympass/yoga",
  "version": "5.2.0-alpha.4",
  "description": "Gympass component library",
  "main": "./src",
  "sideEffects": false,
  "scripts": {
    "build": "yarn build:web && yarn build:native && yarn build:esm",
    "build:web": "NODE_ENV=web babel ./src --out-dir dist/cjs",
    "build:esm": "NODE_ENV=esm babel ./src --out-dir dist/esm",
    "build:native": "NODE_ENV=native babel ./src --out-dir dist/cjs",
    "prebuild": "rm -rf ./dist",
    "postbuild": "cp -r ./src/Fonts ./dist/fonts",
    "prepublishOnly": "node ../../scripts/prepublish.js --rn"
  },
  "keywords": [
    "Gympass",
    "components",
    "styled-components",
    "react",
    "design-system"
  ],
  "homepage": "https://github.com/Gympass/yoga#readme",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "directory": "dist"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Gympass/yoga.git"
  },
  "bugs": {
    "url": "https://github.com/Gympass/yoga/issues"
  },
  "dependencies": {
    "@gympass/yoga-common": "^0.1.2-alpha.4",
    "@gympass/yoga-helpers": "^0.1.1-alpha.0",
    "@gympass/yoga-icons": "^0.11.2-alpha.5",
    "@gympass/yoga-tokens": "^1.1.0-alpha.0",
    "@ptomasroos/react-native-multi-slider": "^2.2.2",
    "downshift": "^6.1.0",
    "prop-types": "^15.7.2",
    "proxy-polyfill": "^0.3.2",
    "rc-slider": "^9.7.1",
    "react-google-font-loader": "^1.1.0"
  },
  "devDependencies": {
    "@testing-library/react": "^11.2.5",
    "@testing-library/react-native": "^7.2.0",
    "babel-plugin-inline-react-svg": "^2.0.1",
    "styled-components": "^5.2.1"
  },
  "peerDependencies": {
    "styled-components": "^5.2.1"
  }
}
