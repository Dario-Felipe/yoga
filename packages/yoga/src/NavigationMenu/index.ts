import WebNavigationMenu from './web';

export default WebNavigationMenu;
