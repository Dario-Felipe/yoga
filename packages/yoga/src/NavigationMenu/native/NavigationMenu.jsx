const NavigationMenu = () => null;

export default NavigationMenu;
