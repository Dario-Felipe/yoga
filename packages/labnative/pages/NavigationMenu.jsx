import React from 'react';
import { View } from 'react-native';
import { NavigationMenu } from '@gympass/yoga';

const NavigationMenuPage = () => (
  <View>
    <NavigationMenu>NavigationMenu</NavigationMenu>
  </View>
);

export default NavigationMenuPage;
