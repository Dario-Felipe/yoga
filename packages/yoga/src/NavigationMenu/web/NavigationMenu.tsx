import React from 'react';
import Menu from './Menu';
import Switcher from './Switcher';
import { Item, SubItem } from './Item';

import * as Styles from './NavigationMenu.styles';

type NavigationMenuProps = {
  children: React.ReactNode;
};

function NavigationMenu({ children }: NavigationMenuProps): JSX.Element {
  return <Styles.NavigationMenu>{children}</Styles.NavigationMenu>;
}

NavigationMenu.Header = Styles.Header;
NavigationMenu.Menu = Menu;
NavigationMenu.Switcher = Switcher;
NavigationMenu.Items = Styles.Items;
NavigationMenu.ItemsGroup = Styles.ItemsGroup;
NavigationMenu.Item = Item;
NavigationMenu.SubItem = SubItem;
NavigationMenu.Footer = Styles.Footer;

export default NavigationMenu;
