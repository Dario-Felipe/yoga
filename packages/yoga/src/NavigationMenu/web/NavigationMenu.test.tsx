import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider, Avatar, Icon } from '@gympass/yoga';
import { Help, Doc } from '@gympass/yoga-icons';

import NavigationMenu from './NavigationMenu';

type IconProps = {
  icon: React.ReactElement<typeof Icon>;
};

describe('<NavigationMenu />', () => {
  describe('Snapshots', () => {
    it('should match  NavigationMenu', () => {
      const renderWrapper = ({ children }) => {
        return <a href="https://www.gympass.com">{children}</a>;
      };

      const IconComponent = ({ icon }: IconProps) => {
        return <Icon as={icon} />;
      };

      const itemGroups = [
        [
          {
            expanded: true,
            icon: Doc,
            label: 'Subscription',
            tag: 'new',
            subitems: [{ label: 'Details' }],
          },
          {
            active: true,
            icon: Doc,
            label: 'Billing',
          },
        ],
        [
          {
            icon: Help,
            label: 'Help',
          },
        ],
      ];

      const { container } = render(
        <ThemeProvider>
          <NavigationMenu responsive={false}>
            <NavigationMenu.Header>
              <NavigationMenu.Menu
                avatar={<Avatar.Circle />}
                title="Company"
                subtitle="Reseller"
              />
            </NavigationMenu.Header>

            <NavigationMenu.Items>
              {itemGroups.map(group => (
                <NavigationMenu.ItemsGroup>
                  {group.map(item => (
                    <NavigationMenu.Item
                      active={item.active}
                      expanded={item.expanded}
                      icon={<IconComponent icon={item.icon} />}
                      label={item.label}
                      responsive={false}
                      wrapper={renderWrapper}
                      tag={item.tag}
                    >
                      {item.subitems &&
                        item.subitems.map(({ label }) => (
                          <NavigationMenu.Subitem
                            label={label}
                            wrapper={renderWrapper}
                          />
                        ))}
                    </NavigationMenu.Item>
                  ))}
                </NavigationMenu.ItemsGroup>
              ))}
            </NavigationMenu.Items>

            <NavigationMenu.Footer>
              <NavigationMenu.Switcher
                avatar={<Avatar.Circle />}
                title="User"
                subtitle="Admin, Supervisor"
              />
            </NavigationMenu.Footer>
          </NavigationMenu>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match NavigationMenu.Menu with an action', () => {
      const { container } = render(
        <ThemeProvider>
          <NavigationMenu responsive={false}>
            <NavigationMenu.Header>
              <NavigationMenu.Menu
                avatar={<Avatar.Circle />}
                title="Company"
                subtitle="Reseller"
                onClick={() => null}
              />
            </NavigationMenu.Header>
          </NavigationMenu>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });

    it('should match NavigationMenu.Switcher with actions', () => {
      const actions = [
        {
          id: 'Action 1',
          label: 'Action 1',
          onClick: () => null,
        },
        {
          id: 'Action 2',
          label: 'Action 2',
          onClick: () => null,
        },
      ];

      const { container } = render(
        <ThemeProvider>
          <NavigationMenu responsive={false}>
            <NavigationMenu.Footer>
              <NavigationMenu.Switcher
                actions={actions}
                avatar={<Avatar.Circle />}
                title="User"
                subtitle="Admin, Supervisor"
              />
            </NavigationMenu.Footer>
          </NavigationMenu>
        </ThemeProvider>,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
