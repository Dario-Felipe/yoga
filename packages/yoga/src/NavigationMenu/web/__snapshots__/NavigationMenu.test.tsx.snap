// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationMenu /> Snapshots should match NavigationMenu 1`] = `
.c4 {
  border-radius: 9999px;
  background-color: #9898A6;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
  width: 48px;
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c5 {
  width: 50%;
  height: 50%;
}

.c14 {
  width: 12px;
  height: 12px;
}

.c27 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c15 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 24px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-size: 16px;
}

.c18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-style: solid;
  padding: 4px 8px 4px 8px;
  color: #6B6B78;
  border-radius: 4px;
  border-width: 1px;
  border-color: #9898A6;
  font-size: 12px;
  line-height: 16px;
  font-weight: 500;
}

.c2 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  grid-template-columns: max-content 1fr max-content;
  gap: 8px;
  padding: 12px;
  background-color: #FFFFFF;
  border-radius: 9999px;
  min-height: 64px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
  overflow: hidden;
}

.c7 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  font-size: 14px;
  line-height: 14px;
  font-weight: 500;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c8 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c3 > div:first-child {
  height: 32px;
  width: 32px;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  gap: 8px;
  padding: 12px;
  border-radius: 8px;
}

.c24 > div:first-child {
  height: 32px;
  width: 32px;
}

.c25 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c26 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-weight: 500;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  gap: 8px;
  padding: 10px 8px 10px 16px;
  border-radius: 8px;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  gap: 8px;
  padding: 10px 8px 10px 44px;
  border-radius: 8px;
}

.c20 {
  text-transform: uppercase;
  background-color: #231B22;
  color: #FFFFFF;
  border-radius: 9999px;
}

.c17 {
  color: #6B6B78;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c11 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  background-color: transparent;
  border-radius: 8px;
  list-style-type: none;
  cursor: pointer;
  margin-right: 8px;
}

.c11 svg {
  height: 20px;
  width: 20px;
}

.c11:hover,
.c11:focus {
  background-color: #D7D7E0;
}

.c11:hover .c16,
.c11:focus .c16 {
  color: #231B22;
}

.c11:hover svg,
.c11:focus svg {
  fill: #231B22;
}

.c11 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11 a:hover,
.c11 a:focus {
  outline: none;
}

.c11 a:hover .c12,
.c11 a:focus .c12 {
  background-color: #D7D7E0;
  color: #231B22;
}

.c11 a:hover .c16,
.c11 a:focus .c16 {
  color: #231B22;
}

.c11 a:hover svg,
.c11 a:focus svg {
  fill: #231B22;
}

.c11 a.active {
  cursor: auto;
}

.c11 a.active > .c12 {
  background-color: #FFEEF2;
}

.c11 a.active > .c12 .c16 {
  color: #D8385E;
  font-weight: 500;
}

.c11 a.active > .c12 svg {
  fill: #D8385E;
}

.c11 a.active > .c12 > .c19 {
  background-color: #231B22;
}

.c22 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  background-color: transparent;
  border-radius: 8px;
  list-style-type: none;
  cursor: pointer;
  margin-right: 8px;
}

.c22 svg {
  height: 20px;
  width: 20px;
}

.c22 a {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: auto;
}

.c22 a:hover,
.c22 a:focus {
  outline: none;
}

.c22 a:hover .c12,
.c22 a:focus .c12 {
  background-color: #D7D7E0;
  color: #231B22;
}

.c22 a:hover .c16,
.c22 a:focus .c16 {
  color: #231B22;
}

.c22 a:hover svg,
.c22 a:focus svg {
  fill: #231B22;
}

.c22 a.active {
  cursor: auto;
}

.c22 a.active > .c12 {
  background-color: #FFEEF2;
}

.c22 a.active > .c12 .c16 {
  color: #D8385E;
  font-weight: 500;
}

.c22 a.active > .c12 svg {
  fill: #D8385E;
}

.c22 a.active > .c12 > .c19 {
  background-color: #231B22;
}

.c22 a > .c12 {
  background-color: #FFEEF2;
}

.c22 a > .c12 .c16 {
  color: #D8385E;
  font-weight: 500;
}

.c22 a > .c12 svg {
  fill: #D8385E;
}

.c22 a > .c12 > .c19 {
  background-color: #231B22;
}

.c28 {
  position: relative;
}

.c29 {
  position: absolute;
  bottom: 0;
  display: grid;
  grid-auto-columns: minmax(0,1fr);
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 56px;
  background-color: #FFFFFF;
  margin: 0;
  padding: 0;
  border: 1px solid #D7D7E0;
  gap: 4px;
  z-index: 2;
}

.c33 {
  margin: 0;
  padding: 0;
  font-size: 10px;
  line-height: 12px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
  max-width: 100%;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c30 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  border-radius: 8px;
  list-style-type: none;
  cursor: pointer;
}

.c30 svg {
  height: 20px;
  width: 20px;
  fill: #6B6B78;
}

.c30:hover,
.c30:focus {
  background-color: #D7D7E0;
}

.c30:hover .c32,
.c30:focus .c32 {
  color: #231B22;
}

.c30:hover svg,
.c30:focus svg {
  fill: #231B22;
}

.c30 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c30 a.active {
  cursor: auto;
  background-color: #FFEEF2;
}

.c30 a.active:not(:last-child) {
  pointer-events: none;
}

.c30 a.active div .c32 {
  color: #D8385E;
  font-weight: 500;
}

.c30 a.active div svg {
  fill: #D8385E;
}

.c34 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  border-radius: 8px;
  list-style-type: none;
  cursor: pointer;
  background-color: #FFEEF2;
}

.c34 svg {
  height: 20px;
  width: 20px;
  fill: #6B6B78;
}

.c34 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c34 a.active {
  cursor: auto;
  background-color: #FFEEF2;
}

.c34 a.active:not(:last-child) {
  pointer-events: none;
}

.c34 a.active div .c32 {
  color: #D8385E;
  font-weight: 500;
}

.c34 a.active div svg {
  fill: #D8385E;
}

.c34:not(:last-child) {
  pointer-events: none;
}

.c34 div .c32 {
  color: #D8385E;
  font-weight: 500;
}

.c34 div svg {
  fill: #D8385E;
}

.c31 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 4px;
  padding: 4px;
  border-radius: 8px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 20px;
  padding: 16px 12px;
  background-color: #F5F5FA;
  width: 280px;
  height: 100%;
  right: 0;
  -webkit-transition: width 300ms ease-in-out;
  transition: width 300ms ease-in-out;
  position: fixed;
  width: 100%;
  height: calc(100% - 56px);
  z-index: 1;
  top: 0;
  right: -100%;
  -webkit-transition: right 300ms ease-in-out;
  transition: right 300ms ease-in-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow-y: auto;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0;
  padding: 0 0 16px 0;
  gap: 4px;
}

.c23 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

@media (min-width:1024px) {
  .c28 {
    display: none;
  }
}

@media (min-width:1024px) {
  .c0 {
    position: relative;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-flex-shrink: 0;
    -ms-flex-negative: 0;
    flex-shrink: 0;
    gap: 20px;
    padding: 16px 12px;
    background-color: #F5F5FA;
    width: 280px;
    height: 100%;
    right: 0;
    -webkit-transition: width 300ms ease-in-out;
    transition: width 300ms ease-in-out;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
            display="flex"
            elevation="small"
            height="48"
            overflow="hidden"
            width="48"
          >
            <svg
              aria-hidden="true"
              class="c5"
              fill="#FFFFFF"
              height="50%"
              width="50%"
            />
          </div>
        </div>
        <div
          class="c6"
        >
          <h4
            class="c7"
          >
            Company
          </h4>
          <p
            class="c8"
          >
            Reseller
          </p>
        </div>
      </div>
    </div>
    <nav
      class="c9"
    >
      <ul
        class="c10"
      >
        <li
          class="c11"
        >
          <a
            href="https://www.gympass.com"
          >
            <div
              class="c12 c13"
            >
              <svg
                aria-hidden="true"
                class="c14"
                height="12"
                width="12"
              />
              <p
                class="c15 c16 c17"
                font-size="medium"
              >
                Subscription
              </p>
              <div
                class="c18 c19 c20"
              >
                new
              </div>
            </div>
          </a>
        </li>
        <li
          class="c11"
        >
          <a
            href="https://www.gympass.com"
          >
            <div
              class="c12 c21"
            >
              <p
                class="c15 c16 c17"
                font-size="medium"
              >
                Details
              </p>
            </div>
          </a>
        </li>
        <li
          class="c22"
        >
          <a
            href="https://www.gympass.com"
          >
            <div
              class="c12 c13"
            >
              <svg
                aria-hidden="true"
                class="c14"
                height="12"
                width="12"
              />
              <p
                class="c15 c16 c17"
                font-size="medium"
              >
                Billing
              </p>
            </div>
          </a>
        </li>
      </ul>
      <ul
        class="c10"
      >
        <li
          class="c11"
        >
          <a
            href="https://www.gympass.com"
          >
            <div
              class="c12 c13"
            >
              <svg
                aria-hidden="true"
                class="c14"
                height="12"
                width="12"
              />
              <p
                class="c15 c16 c17"
                font-size="medium"
              >
                Help
              </p>
            </div>
          </a>
        </li>
      </ul>
    </nav>
    <div
      class="c23"
    >
      <div
        class="c24"
        fill="transparent"
      >
        <div
          class="c4"
          display="flex"
          elevation="small"
          height="48"
          overflow="hidden"
          width="48"
        >
          <svg
            aria-hidden="true"
            class="c5"
            fill="#FFFFFF"
            height="50%"
            width="50%"
          />
        </div>
        <div
          class="c25"
        >
          <p
            class="c26"
          >
            User
          </p>
          <p
            class="c27"
            color="deep"
          >
            Admin, Supervisor
          </p>
        </div>
      </div>
    </div>
  </div>
  <nav
    class="c28"
  >
    <ul
      class="c29"
    >
      <li
        class="c30"
      >
        <a
          href="https://www.gympass.com"
        >
          <div
            class="c31"
          >
            <svg
              aria-hidden="true"
              class="c14"
              height="12"
              width="12"
            />
            <p
              class="c32 c33"
            >
              Subscription
            </p>
          </div>
        </a>
      </li>
      <li
        class="c34"
      >
        <a
          href="https://www.gympass.com"
        >
          <div
            class="c31"
          >
            <svg
              aria-hidden="true"
              class="c14"
              height="12"
              width="12"
            />
            <p
              class="c32 c33"
            >
              Billing
            </p>
          </div>
        </a>
      </li>
      <li
        class="c30"
      >
        <div>
          <div
            class="c31"
          >
            <svg
              aria-hidden="true"
              class="c14"
              height="12"
              width="12"
            />
            <p
              class="c32 c33"
            >
              More
            </p>
          </div>
        </div>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`<NavigationMenu /> Snapshots should match NavigationMenu.Menu with an action 1`] = `
.c4 {
  border-radius: 9999px;
  background-color: #9898A6;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
  width: 48px;
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c5 {
  width: 50%;
  height: 50%;
}

.c10 {
  width: 24px;
  height: 24px;
}

.c2 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  grid-template-columns: max-content 1fr max-content;
  gap: 8px;
  padding: 12px;
  background-color: #FFFFFF;
  border-radius: 9999px;
  min-height: 64px;
  cursor: pointer;
}

.c2:hover,
.c2:focus {
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
  overflow: hidden;
}

.c7 {
  margin: 0;
  padding: 0;
  font-size: 24px;
  line-height: 32px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 500;
  font-size: 14px;
  line-height: 14px;
  font-weight: 500;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c8 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c9 svg {
  fill: #D8385E;
}

.c3 > div:first-child {
  height: 32px;
  width: 32px;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 20px;
  padding: 16px 12px;
  background-color: #F5F5FA;
  width: 280px;
  height: 100%;
  right: 0;
  -webkit-transition: width 300ms ease-in-out;
  transition: width 300ms ease-in-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <div
            class="c4"
            display="flex"
            elevation="small"
            height="48"
            overflow="hidden"
            width="48"
          >
            <svg
              aria-hidden="true"
              class="c5"
              fill="#FFFFFF"
              height="50%"
              width="50%"
            />
          </div>
        </div>
        <div
          class="c6"
        >
          <h4
            class="c7"
          >
            Company
          </h4>
          <p
            class="c8"
          >
            Reseller
          </p>
        </div>
        <div
          class="c9"
        >
          <svg
            aria-hidden="true"
            class="c10"
            height="24"
            width="24"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<NavigationMenu /> Snapshots should match NavigationMenu.Switcher with actions 1`] = `
.c3 {
  border-radius: 9999px;
  background-color: #9898A6;
  box-shadow: 0 2px 4px -1px rgba(152,152,166,0.2),0 4px 5px 0px rgba(152,152,166,0.14),0 1px 10px 0px rgba(152,152,166,0.12);
  width: 48px;
  height: 48px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
}

.c4 {
  width: 50%;
  height: 50%;
}

.c9 {
  width: 20px;
  height: 20px;
}

.c7 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  color: #6B6B78;
}

.c8 {
  -webkit-transition: background-color 300ms ease-in-out;
  transition: background-color 300ms ease-in-out;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 40px;
  height: 40px;
  border-radius: 9999px;
}

.c8 svg {
  fill: #D8385E;
}

.c8:hover,
.c8:focus {
  cursor: pointer;
  background-color: #D7D7E0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
  gap: 8px;
  padding: 12px;
  border-radius: 8px;
}

.c2 > div:first-child {
  height: 32px;
  width: 32px;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: hidden;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  color: #231B22;
  font-family: Rubik;
  font-weight: 400;
  font-weight: 500;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
}

.c0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  gap: 20px;
  padding: 16px 12px;
  background-color: #F5F5FA;
  width: 280px;
  height: 100%;
  right: 0;
  -webkit-transition: width 300ms ease-in-out;
  transition: width 300ms ease-in-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        fill="transparent"
      >
        <div
          class="c3"
          display="flex"
          elevation="small"
          height="48"
          overflow="hidden"
          width="48"
        >
          <svg
            aria-hidden="true"
            class="c4"
            fill="#FFFFFF"
            height="50%"
            width="50%"
          />
        </div>
        <div
          class="c5"
        >
          <p
            class="c6"
          >
            User
          </p>
          <p
            class="c7"
            color="deep"
          >
            Admin, Supervisor
          </p>
        </div>
        <div
          aria-expanded="false"
          aria-haspopup="menu"
          class="c8"
          data-state="closed"
          id="radix-:r0:"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="c9"
            height="20"
            width="20"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
